### ISSUE TITLE
Gaussian Shell Maps for Efficient 3D Human Generation

### ISSUE THREAD

**Issue by w-m** [w-m/awesome-3d-splatting-survey#7]
## TLDR

[EG3D](https://nvlabs.github.io/eg3d/) + [3DGS](#1) for human bodies. Gaussians live in fixed positions on 8 "shells" around the SMPL surface in T-Pose.

<img width="1015" alt="Bildschirmfoto 2023-12-05 um 12 20 03" src="https://github.com/w-m/awesome-3d-splatting-survey/assets/1588593/0ed769fd-f39d-4804-8484-bfc51cf89f6f">

### Main Idea or Main Points
- Modelling near-surface details by using 8 shells around the SMPL mesh surface.
- The shells are created with Laplacian Smoothing on the SMPL mesh, with the smoothing factor set to negative for inflation and positive for deflation.
- Stores Gaussian parameters as texture maps, for each shell independently.

### Novelty
- Somewhat straightorward combination of EG3D + 3DGS
- Shell Maps (from 2005) used for Gaussians. They are used for NeRFs in some 2023 publications? TODO: check [14, 47, 58] refs in the paper.
- A custom scaling regularization: "In our empirical study, we observe when unconstrained, the network tends to learn overly large or extremely small Gaussians early on during the training and rapidly leads to divergence or model collapse. We experimented with multiple regularization strategies and different activation functions and found the following scaling regularization: ..."

### Areas of Improvement / Problems
- Struggles with dynamic elements like hair and clothing. Animated/articulated T-Posed Gaussian person doesn't look very natural.
- Complexity in generalizing over a distribution of objects/scenes due to Gaussian parameter correlations.
- Problems with convergence:
	- tetrahedron variant has worse FID than triangle variant
	- 200k Gaussians have worse FID than 100k Gaussians
- FID worse than AG3D for Deep Fashion

### Open Questions / Reader's Questions / Clarifications Needed
- Goes into detail about tetrahedra forming shell space. But it seems they finally use only Gaussians living on the surface (shell), not between them. Why didn't the tetrahedron idea not work out?
- How are the shells formed? Does the Laplacian Smoothing only create the bounds, and the shells are interpolated between? How? Or are the shells formed from different Laplacian params?
- What are the Laplacian params to form the shells?
- "The offset between adjacent shells is 0.08" - is this a parameter or a distance?
- Lacking details and visualizations of the shells and their barycentric coordinates (just in the surface, or above, interpolation?) and the sampling of the Gaussians.
- Scaling regularization: the scale is regularized to the average distance of the closest Gaussian neighbor. It is not explained further, why this particular choice is a good idea.
- How much does the regularization affect the scale? Do they end up being all the same size?
- TODO check: logarithm in the scaling regularization is to counteract the exponential activation?
- Why are they using only 100k Gaussians, when there are 512x512 (=262k) pixels?

### Future Directions
- "Due to the irregularity and sparsity of Gaussians, it is not obvious how to extract the accurate geometry and normals. Surface splatting, with a stronger surface prior may be a potential solution, which we plan to investigate in the future."
- Authors suggest to combine studio captures with in-the-wild datasets for richer human appearance modeling.

## Paper content

The following is mostly courtesy of GPT-4, with minor edits.

### Problem Solved
- Addresses the challenge of efficiently generating 3D digital humans, crucial for various industries.

### Results
- SOTA rendering speed in single-view datasets like SHHQ and DeepFashion.
- Generates detailed human figures, including complex features like clothing and accessories.
- Achieves high-quality, multi-view consistent renderings without upsamplers.

### Comparative Analysis
- Surpasses existing methods in rendering speed and multi-view consistency.
- Avoids post-rendering convolution layers, eliminating aliasing artifacts common in other methods.

### Methodology / Approach
- Utilizes shell maps for texture mapping and detail modeling.
- Adopts StyleGAN2 for the generator and a customized discriminator for RGB and alpha channels.
- Dedicated discriminators for face, hands+feet

### Experimental Setup
- Trained on datasets like DeepFashion and SHHQ.
- Uses 8 shells with 100k Gaussians, trained on 8 A100 GPUs.

## Links and facts

| Category                 | Details                    |
|--------------------------|----------------------------|
| First Author             | Rameen Abdal               |
| Last Author              | Gordon Wetzstein           |
| Author Organizations     | Stanford University, HKUST |
| Dataset Used             | DeepFashion, SHHQ          |
| ArXiv Link               | [arXiv:2311.17857v1](https://arxiv.org/abs/2311.17857v1) |
| Project Page             | https://rameenabdal.github.io/GaussianShellMaps/ |
| Code Link                | https://github.com/computational-imaging/GSM|
| Submission Date          | 29 Nov 2023                |
| Keywords                 | 3D Human Generation, Gaussian Shell Maps, GANs, Rendering |
| Acknowledgements         | Google, Samsung, Stanford HAI, Swiss Postdoc Mobility Fund.

## Visualization / Illustrations

<img width="508" alt="Bildschirmfoto 2023-12-05 um 12 13 41" src="https://github.com/w-m/awesome-3d-splatting-survey/assets/1588593/e5c4fc28-8769-46a2-a218-d2ced2e47736">

## Abstract

Efficient generation of 3D digital humans is important in several industries, including virtual reality, social media, and cinematic production. 3D generative adversarial networks (GANs) have demonstrated state-of-the-art (SOTA) quality and diversity for generated assets. Current 3D GAN architectures, however, typically rely on volume representations, which are slow to render, thereby hampering the GAN training and requiring multi-view-inconsistent 2D upsamplers. Here, we introduce Gaussian Shell Maps (GSMs) as a framework that connects SOTA generator network architectures with emerging 3D Gaussian rendering primitives using an articulable multi shell–based scaffold. In this setting, a CNN generates a 3D texture stack with features that are mapped to the shells. The latter represent inflated and deflated versions of a template surface of a digital human in a canonical body pose. Instead of rasterizing the shells directly, we sample 3D Gaussians on the shells whose attributes are encoded in the texture features. These Gaussians are efficiently and differentiably rendered. The ability to articulate the shells is important during GAN training and, at inference time, to deform a body into arbitrary userdefined poses. Our efficient rendering scheme bypasses the need for view-inconsistent upsamplers and achieves highquality multi-view consistent renderings at a native resolution of 512 × 512 pixels. We demonstrate that GSMs successfully generate 3D humans when trained on single-view  datasets, including SHHQ and DeepFashion.
